ospfd/ospf_lsa.c

```diff
@@ -2339,6 +2371,23 @@ struct ospf_lsa *ospf_external_lsa_refresh(struct ospf *ospf,
 
 	new->data->ls_seqnum = lsa_seqnum_increment(lsa);
 
+	/* Check if any content actually changed */
+	if (LS_AGE(lsa) < OSPF_LS_REFRESH_TIME - (2 * OSPF_LS_REFRESH_JITTER) &&
+	    ospf_lsa_same(lsa, new)) {
```

Comment:
OSPF external LSA refreshed when new route added or changed values from zebra. ospf_redistribute_check() checks if LSA has changed or not along with routermap filter. There is force flag passed, i guess in that case it is forced to do refresh. Have you checked force flag value LSA_REFRESH_FORCE or LSA_REFRESH_IF_CHANGED can be employed from the caller? Have you exhust the cases where FORCE flag is passed which meant to do force refresh? 
Have you done a UT where existing external route added or modified triggers a refresh even tough AGE within refresh window?

Reply:
LSA_REFRESH_FORCE probably needs to be handled here -- just add a check that if that flag is set, refresh even though ospf_lsa_same returns true, I think